x-pack/metricbeat/module/aws/aws.go

```diff
@@ -184,6 +185,14 @@ func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
 		awsConfig.Region = config.Regions[0]
 	}
 
+	// Starting from Go 1.24, when FIPS 140-3 mode is active, fips140.Enabled() will return true.
+	// So, regardless of whether `fips_enabled` is set to true or false, when FIPS 140-3 mode is active, the
+	// resolver will resolve to the FIPS endpoint.
+	// See: https://go.dev/doc/security/fips140#fips-140-3-mode
+	if fips140.Enabled() {
+		config.AWSConfig.FIPSEnabled = true
+	}
+
```

Comment:
Shall we move this configuration setting just below the configuration unpack [1] ? This makes sure unpacked configuration is updated before the first use. (this is already done in other files)

[1] - https://github.com/elastic/beats/blob/fc621f107f190a45bee95e63c27b7aaffb044ff7/x-pack/metricbeat/module/aws/aws.go#L124-L128

Reply:
Sure.

Reply:
Done.